docker/DockerfileNoetic

```diff
@@ -0,0 +1,77 @@
+# Start with ROS base image
+FROM ros:noetic-ros-base
+
+# Tools useful during development.
+RUN apt update \
+ && apt install -y --no-install-recommends\
```

Comment:
```suggestion
 && apt install -y --no-install-recommends\
              bash \
```

Reply:
@j-herman Thanks for all these notes. 

## host-image OS differences
I think you found a few things that are specific to users who are running a different OS on the host. This should work, but since you are using the --user and --home options of the rocker command you are mounting your local home directory which pulls in the configuration files for bionic/melodic/gaz9, and your workspace built for that OS. 
* Just curious: if you run `catkin_make clean` instead of manually deleting the build and devel folders, does that work?
* It seems like using the same .bashrc is not a good plan unless the host and image are very similar. We can add a bashrc specific to the image instead.
* To answer your question about the rocker installation, I would say a virtual environment is recommended but not required.

## `join.bash`
I verified that `bash` is installed on the container. To see this, run:
```
which bash
```
in the container terminal. It will show you bash it at `/usr/bin/bash`. This is also the reason your command `docker exec -it <container ID> /bin/bash` works. That command executes the file at `/bin/bash` (which is a symbol link to `/usr/bin/bash`) _inside_ the container. If `bash` were not installed in the container it would fail.

The problem you encountered is that the tutorial gives you the wrong argument for the `join.bash` script: `dave_noetic`. This is the name we gave the first image you built, but we forgot that `rocker` creates another image from this one and uses a default random hash for the name. This second image is the one that gets launched as a container (i.e. running version of an image).

The result is that so you are trying to join a container that isn't running at all (and therefore also not running `bash`). The quickest fix to this is to edit the tutorial to tell the user to tag the rocker image after it gets created. Longer term, we should probably fix up our `join.bash` script, because it is kind of hacky (it asks for the name of the image and tries to look up the name of the running container that was generated from that image, but this is not ideal because it is possible to run multiple containers from the same base image, and if someone were doing that the script would not be able to distinguish between them).

Reply:
That makes much more sense!  I thought I had run which bash but going back through my full notes I can't find it.  Deleted my non-functioning suggestion and will revise my notes on the pull request.  catkin_make clean does not work correctly on my system (no idea why, been like that through many updates and reinstalls) but deleting build and devel does the same thing, right?

Reply:
`catkin_make clean` is not quite as thorough as `rm -rf build/ devel/`.  I thought it might be good enough, but apparently not. Thanks for checking.